Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -235,30 +268,55 @@ void __stdcall PlayerLaunch(unsigned int iShip, unsigned int client)
 		// Check to see if the carrier is currently in a base. If so, force the client to dock on that base.
 		if(!carrierShip)
 		{
-
 			uint iBaseID;
 			pub::Player::GetBase(carrier_client, iBaseID);
 
-			if(!iBaseID)
+			if (iBaseID)
 			{
-				return;
+				// Set the flags which the PlayerLaunch_AFTER uses to handle teleporting to the carriers base
+				mobiledockClients[client].undockBase = Universe::get_base(iBaseID);
+				mobiledockClients[client].baseUndock = true;
 			}
+			else
+			{
+				// If died carrier and docked ship are in same system.

```

Comment:
died carrier?

Reply:
If ship doesn't exists and client is not at base, then probably client is watching death screen.
May bug up if carrier respawns before docked ships are launched in space.

Reply:
Fixed.